src/generators/clientFileGenerator.ts

```diff
@@ -559,10 +560,11 @@ function writeDefaultOptions(
   hasLro: boolean,
   clientDetails: ClientDetails
 ) {
-  const { useCoreV2, credentialScopes, packageDetails } = getAutorestOptions();
+  const { useCoreV2, packageDetails } = getAutorestOptions();
+  const { credentialScopes } = getSecurityInfoFromModel(clientDetails.security);
 
   const credentialScopesValues = getCredentialScopesValue(credentialScopes);
-  const addScopes = credentialScopes
+  const addScopes = credentialScopes && credentialScopes.length > 0
```

Comment:
Here we can check if we have addCredentials and the old settings available in `getAutorestOptions` if so, use them otherwise use the settings from `getSecurityInfoFromModel`